packages/kotti-ui/source/props.ts

```diff
@@ -0,0 +1,198 @@
+
+			const options = shapePointer._def.options as Array<z.ZodTypeAny>
+
+			// walk every option in the enum and merge result sets into types
+			for (const typeSet of options.map((x) => walkShape(x)))
+				for (const type of typeSet) types.add(type)
+
+			if (DEBUG_WALK_SHAPE) console.groupEnd()
+		} else {
+			if (DEBUG_WALK_SHAPE) console.log(`walkTypes: pushing “${typeName}”`)
+			types.add(typeName)
+		}
+
+		shapePointer = shapePointer._def.innerType
+	}
+
+	return types
+	/* eslint-enable no-console */
+}
+
+/**
+ * These types do not influence the generation of Vue’s `propName.type` Arrays and are therefore discarded
+ */
+const ignoredZodTypes = new Set(['ZodDefault', 'ZodNullable', 'ZodOptional'])
+
+/**
+ * This maps the internal zod name of a type to the the constructor that Vue expects in `propName.type`
+ */
+const zodToVueTypes = new Map<
+	string,
+	Exclude<PropType<unknown>, Array<unknown>>
+>([
+	['ZodArray', Array],
+	['ZodBoolean', Boolean],
+	['ZodFunction', Function],
+	['ZodNativeEnum', String], // educated guess, can be fixed if the need for non-string enums arises
+	['ZodNumber', Number],
+	['ZodObject', Object],
+	['ZodString', String],
+])
+
+/**
+ * @summary Easily define & validate vue props, improve error messages drastically
+ *
+ * @description
+ * This augments/replaces Vue’s props entirely, including the need to add prop.validator
+ * by using zod and therefore actually prints errors to the console that can help developers
+ * rather than just say "hey you failed the validator, good luck!" as vue would do by default.
+ *
+ * ## Known limitations:
+ *
+ * 1. Deeply defined defaults are only applied when manually `schema.safeParse`-ing the data (only top-level defaults are possible to transform to Vue)
+ * 2. While this function also automatically types the return types via as PropType<> (which in theory
+ *   should mean that annotating the prop types explicitly is no longer necessary) in `@vue/composition-api`
+ *   (`0.6.1` and `1.1.5`), `PropType<>` appears to be bugged.
+ *   For example KtBreadcrumb’s props get inferred to be any
+ *   So, unfortunately, this may only help in the future, once the Vue team fixes this
+ * 3. `z.function().default()` does not accept `undefined` while parsing the schema manually
+ *   without an explicit `.optional()` {@link https://github.com/colinhacks/zod/issues/647}
+ *   However, it can still be used to generate the Vue defaults, as Vue doesn’t call validators with `undefined`
+ *
+ * @example
+ * // KtUserMenu.vue
+ * export default defineComponent<KottiUserMenu.PropsInternal>({
+ *   name: 'KtUserMenu',
+ *   props: makeProps(KottiUserMenu.propsSchema)
+ * })
+ *
+ * // types.ts
+ * export namespace KottiUserMenu {
+ *   const propsSchema = z.object({
+ *     // ...
+ *   })
+ *   type Props = z.input<typeof propsSchema>
+ *   type PropsInternal = z.output<typeof propsSchema>
+ * }
+ */
+export const makeProps = <SCHEMA extends z.ZodObject<z.ZodRawShape>>(
```

Comment:
I think I can verify that:
1) `z.ZodObject<z.ZodRawShape>` is equivalent to
2) `z.ZodRawShape`

in our case, because this is somehow only useful if we use recursive schemas IMU

Reply:
from all the prop type defintions, you're using `z.object` constructor, so I will not tire myself with understanding this further

since an object constructor simply should return a `ZodObject` as a wrapping type for each of the props type 

nevertheless:

the generic passed to the object being a rawshape, means that each of the props is an object and not a direct type/(schema in the terms of zod)


will see how it's used further then resolve this comment if fit